agent.py

```diff
@@ -29,15 +29,23 @@ def __init__(self, args, env):
                     args.model, map_location='cpu'
                 )  # Always load tensors onto CPU by default, will shift to GPU if necessary
                 if 'conv1.weight' in state_dict.keys():
-                    for old_key, new_key in (('conv1.weight',
-                                              'convs.0.weight'),
-                                             ('conv1.bias', 'convs.0.bias'),
-                                             ('conv2.weight',
-                                              'convs.2.weight'),
-                                             ('conv2.bias', 'convs.2.bias'),
-                                             ('conv3.weight',
-                                              'convs.4.weight'),
-                                             ('conv3.bias', 'convs.4.bias')):
+                    for old_key, new_key in (
+                        ('conv1.weight', 'convs.0.weight'),
+                        ('conv1.bias', 'convs.0.bias'),
+                        ('conv2.weight', 'convs.2.weight'),
+                        ('conv2.bias', 'convs.2.bias'),
+                        ('conv3.weight', 'convs.4.weight'),
+                        ('conv3.bias', 'convs.4.bias')):
+                        state_dict[new_key] = state_dict[
+                            old_key]  # Re-map state dict for old pretrained models
+                        del state_dict[
+                            old_key]  # Delete old keys for strict load_state_dict
+                if 'convs.0.weight' in state_dict.keys():
+                    for old_key, new_key in (
+                        ('convs.0.weight', 'phi.0.weight'),
+                        ('convs.0.bias', 'phi.0.bias'),
+                        ('convs.2.weight', 'phi.2.weight'),
+                        ('convs.2.bias', 'phi.2.bias')):
```

Comment:
This was a bit of a hack. When you change the names of the variables (e.g. `convs` -> `phi`) it changes the names of the corresponding weights that get saved. The network I loaded used `convs`, but I was loading it into a network that used `phi` and thus things broke.

Since OP already had code to fix this problem from a previous renaming, I just extended it for our purposes. However, since we are training RainbowDE, there's one fewer layer in the cnn, which meant there were no `phi.4.*` weights to set in the `state_dict`

Reply:
Maybe wrap both in different functions? No real need tbh.